packages/fast-usdc/src/exos/liquidity-pool.js

```diff
@@ -178,7 +182,25 @@ export const prepareLiquidityPoolKit = (zone, zcf, USDC, tools) => {
           Object.assign(this.state, post);
           this.facets.external.publishPoolMetrics();
         },
-        // TODO method to repay failed `LOA.deposit()`
+        /**
+         * If something fails during advance, return funds to the pool.
+         *
+         * @param {ZCFSeat} borrowSeat
+         * @param {{ USDC: Amount<'nat'>}} amountKWR
+         */
+        repay(borrowSeat, amountKWR) {
+          const { zcfSeat: repaySeat } = zcf.makeEmptySeatKit();
+          const returnAmounts = harden({
+            Principal: amountKWR.USDC,
+            PoolFee: makeEmpty(USDC),
+            ContractFee: makeEmpty(USDC),
+          });
+          // arrange payments in a format repay is expecting
+          zcf.atomicRearrange(
+            harden([[borrowSeat, repaySeat, amountKWR, returnAmounts]]),
+          );
```

Comment:
I don't think we need try/catch here since we are protected by the `borrow.repay` interface guard.

LMK if that thinking is off base and/or a code comment is warranted,

Reply:
The interface guard doesn't prevent calling this function with a `borrowSeat` that doesn't have enough USDC.

Check `isGTE(borrowSeat.getAmountAllocated('USDC'), amountToRepay)` before calling `atomicRearrange`.

And I think we need a try/catch here exactly as much as in the other functions that reallocate. I can see arguments either way (the commit phase is synchronous for all of them), but I don't see how this one should be different.

Reply:
re-reading [Prepare / Commit](https://github.com/Agoric/agoric-sdk/wiki/Prepare-Commit), the `try` / `catch` is only in order _in case the commit phase is not atomic_.

Reply:
> Check isGTE(borrowSeat.getAmountAllocated('USDC'), amountToRepay) before calling atomicRearrange

Agree, added.

> re-reading Prepare / Commit, the try / catch is only in order in case the commit phase is not atomic.

I think this means we should remove the `zcf.shutdownWithFailure(reason);`'s in liquidity-pool? If yes, please see 85ce625f83

Reply:
please remove all the shutdowns;  leave >= 1 UNTIL #10684 comment.

In some cases, the `try` part of the commit has multiple synchronous steps. Part of me was thinking of that as atomic, since it's all 1 turn. But it's not atomic. Regardless, recent discussion suggests removing all `shutdownWithFailure` calls UNTIL #10684.

> see 1d8c44e

I tried; no joy.